lib/workers/bin/run_single_worker.rb

```diff
@@ -23,10 +23,6 @@
     options[:dry_run] = val
   end
 
-  opts.on("-g=GUID", "--guid=GUID", "Find an existing worker record instead of creating") do |val|
-    options[:guid] = val
-  end
```

Comment:
Isn't this useful for development?

Reply:
nvm, I see you are removing CLI params in favor of ENV vars.  If we do that we should probably just remove *ALL* CLI params, so not sure if it makes sense.

Reply:
I am very much against a notion of using ENV vars over CLI flags.

CLI flags are explicit, and can be passed in from other processes, where as ENV vars can be polluted from processes that came before, or copy pasting something in to a `.bashrc` (or equivalent), making it hard to track down bugs (in my own experience).

As to "why I mixed and matched CLI and ENV vars" (asked by @jrafanie here:  https://github.com/ManageIQ/manageiq/pull/16140/files#r143249599 ), I honestly forget, but will be looking that up.  But I explicitly made a choice NOT to configured this script via ENV variables alone for the reasons I just mentioned, and would urge that not being the main mechanism for configuring a work.

Just incase this is not clear/trying to cut another question off at the pass, I am not saying that in containers we don't use ENV vars to configure the workers, but the values of those ENV vars should be what is passed into the flags, like so:

```console
$ ruby lib/workers/run_single_worker_script --guid=$MY_WORKER_GUID ...
```

And not just reading from the `ENV` in the application code, in my option.

Reply:
> why I mixed and matched CLI and ENV vars

I don't think you did.  You preferred having a CLI var.  He is undoing that in this PR.  FWIW, I agree with you that run_single_worker should just support CLI vars only, and containers can use the vars in the entrypoint definition.  This will also affect Procfiles however: https://github.com/ManageIQ/manageiq/blob/6be82739e07be7ab28a6c71b60ca69585deacf00/Procfile.example#L48-L50  IMO, the Procfile should also change to just use CLI vars.

Reply:
Just to add some context (versus opinions/suggestions), here is what the script started as:

https://github.com/ManageIQ/manageiq/pull/15132/files#diff-b11ed4ff48cc4551888048e1d7d4add7

The only ENV var that I did include in there was overridden by a flag, and that was for working with the disabling heartbeating functionality (allowing this to be run in as a single worker, without a needing a MiqServer running for it to work), which was done in a [separate effort](https://github.com/ManageIQ/manageiq/pull/15123) before this script made it in as a PR.

Any ENV manipulation added since then wasn't added by me, I guess (though I forgot that was the case), but added with the addition of bundler groups ( @jrafanie ) and using `foreman` ( @carbonin ), which should have associated PRs with them (too lazy to look them up).

Reply:
I would say "yes", but that kind of contradicts what I have been saying [here](https://github.com/ManageIQ/manageiq/pull/16140#pullrequestreview-67777836) and [here](https://github.com/ManageIQ/manageiq/pull/16140#issuecomment-334865544)...